src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -1311,3 +1329,86 @@ double GetStreamCpuLoad( PaStream * s )
 
     return PaUtil_GetCpuLoad( &stream->cpuLoadMeasurer );
 }
+
+/** Extensions */
+static void RenameStreamCb(pa_stream *s, int success, void *userdata)
+{
+    /* Currently does nothing but signal the caller. */
+    PaPulseAudio_Stream *l_ptrStream = (PaPulseAudio_Stream *) userdata;
+    pa_threaded_mainloop_signal( l_ptrStream->mainloop,
+                                 0 );
+}
+
+PaError PaPulseAudio_RenameSource( PaStream *s, const char *streamName )
+{
+    PaPulseAudio_Stream *stream = (PaPulseAudio_Stream *) s;
+    PaError result = paNoError;
+    pa_operation *op = NULL;
+
+    if ( stream->inStream == NULL )
+    {
+        return paInvalidDevice;
+    }
+
+    /* Reallocate stream name in memory. */
+    pa_threaded_mainloop_lock( stream->mainloop );
+    char *newStreamName = (char*)PaUtil_AllocateMemory(strnlen(streamName, PAPULSEAUDIO_MAX_DEVICENAME) + 1);
+    if ( !newStreamName )
+    {
+        pa_threaded_mainloop_unlock( stream->mainloop );
+        return paInsufficientMemory;
+    }
+    memset(newStreamName, 0, strnlen(streamName, PAPULSEAUDIO_MAX_DEVICENAME) + 1);
+    strncpy(newStreamName, streamName, strnlen(streamName, PAPULSEAUDIO_MAX_DEVICENAME));
```

Comment:
a save and fast way us using snprintf("%s" ...) it does not look like this, but the compiler is smart enough to get rid of the format parsing at runtime.

Reply:
Updated.